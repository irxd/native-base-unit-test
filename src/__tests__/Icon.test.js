import React from 'react';
import { render } from '@testing-library/react-native';
import {
  NativeBaseProvider, createIcon
} from 'native-base';
import { Circle } from "react-native-svg";

describe('<Icon />', () => {
  const CustomIcon = createIcon({
    viewBox: "0 0 36 36",
    path: [<Circle cx="18" cy="18" r="17.5" fill="#06B6D4" stroke="#0E7490" />, <Circle cx="18" cy="18" r="13.5" fill="white" stroke="#0E7490" />, <Circle cx="18" cy="18" r="9.5" fill="#06B6D4" stroke="#0E7490" />, <Circle cx="18" cy="18" r="5.5" fill="white" stroke="#0E7490" />]
  });

  it('should render icon component properly', () => {
    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <CustomIcon size={12} testID="my-icon" />
      </NativeBaseProvider>
    );

    expect(queryByTestId('my-icon')).not.toBeNull();
  });

  it('should render icon props properly', () => {
    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <CustomIcon size={6} testID="my-icon" backgroundColor="black" color="white" />
      </NativeBaseProvider>
    );
    
    expect(queryByTestId('my-icon').props.style[1].backgroundColor).toMatch('#000000');
    expect(queryByTestId('my-icon').props.style[1].color).toMatch('#FFFFFF');
  });
})