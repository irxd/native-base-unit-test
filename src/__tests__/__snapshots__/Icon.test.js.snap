// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render icon props properly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSVGSvgView
    _text={
      Object {
        "color": "lightText",
      }
    }
    accessibilityRole="image"
    align="xMidYMid"
    bbHeight={24}
    bbWidth={24}
    color={4294967295}
    dataSet={Object {}}
    focusable={false}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke=""
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "backgroundColor": "#000000",
          "color": "#FFFFFF",
          "height": 24,
          "width": 24,
        },
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    testID="my-icon"
    tintColor={4294967295}
    vbHeight={36}
    vbWidth={36}
  >
    <RNSVGGroup
      propList={
        Array [
          "stroke",
        ]
      }
      stroke={null}
    >
      <RNSVGGroup>
        <RNSVGCircle
          cx="18"
          cy="18"
          fill={4278630100}
          propList={
            Array [
              "fill",
              "stroke",
            ]
          }
          r="17.5"
          stroke={4279137424}
        />
        <RNSVGCircle
          cx="18"
          cy="18"
          fill={4294967295}
          propList={
            Array [
              "fill",
              "stroke",
            ]
          }
          r="13.5"
          stroke={4279137424}
        />
        <RNSVGCircle
          cx="18"
          cy="18"
          fill={4278630100}
          propList={
            Array [
              "fill",
              "stroke",
            ]
          }
          r="9.5"
          stroke={4279137424}
        />
        <RNSVGCircle
          cx="18"
          cy="18"
          fill={4294967295}
          propList={
            Array [
              "fill",
              "stroke",
            ]
          }
          r="5.5"
          stroke={4279137424}
        />
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
</RNCSafeAreaProvider>
`;
